test/iifInterfaceTests.jl

```diff
@@ -0,0 +1,437 @@
+# using GraphPlot
+# using Neo4j
+# using DistributedFactorGraphs
+# using IncrementalInference
+# using Test
+# dfg = apis[2]
+
+global dfg,v1,v2,f1
+
+if typeof(dfg) <: CloudGraphsDFG
+    @warn "TEST: Nuking all data for user '$(dfg.userId)', robot '$(dfg.robotId)'!"
+    clearRobot!!(dfg)
+end
+
+# Building simple graph...
+@testset "Building a simple Graph" begin
+    global dfg,v1,v2,f1
+    # Use IIF to add the variables and factors
+    v1 = addVariable!(dfg, :a, ContinuousScalar, labels = [:POSE])
+    v2 = addVariable!(dfg, :b, ContinuousScalar, labels = [:LANDMARK])
+    f1 = addFactor!(dfg, [:a; :b], LinearConditional(Normal(50.0,2.0)) )
+    # f1 = addFactor!(fg,[:x0], Prior( pd ) )
+    # @test_throws Exception addFactor!(dfg, DFGFactor{Int, :Symbol}("f2"), [v1, DFGVariable("Nope")])
+end
+
+#test before anything changes
+@testset "Producing Dot Files" begin
+    global dfg
+    todotstr = toDot(dfg)
+    #TODO consider using a regex, but for now test both orders
+    todota = todotstr == "graph graphname {\n2 [\"label\"=\"a\",\"shape\"=\"ellipse\",\"fillcolor\"=\"red\",\"color\"=\"red\"]\n2 -- 3\n3 [\"label\"=\"abf1\",\"shape\"=\"box\",\"fillcolor\"=\"blue\",\"color\"=\"blue\"]\n1 [\"label\"=\"b\",\"shape\"=\"ellipse\",\"fillcolor\"=\"red\",\"color\"=\"red\"]\n1 -- 3\n}\n"
+    todotb = todotstr == "graph graphname {\n2 [\"label\"=\"b\",\"shape\"=\"ellipse\",\"fillcolor\"=\"red\",\"color\"=\"red\"]\n2 -- 3\n3 [\"label\"=\"abf1\",\"shape\"=\"box\",\"fillcolor\"=\"blue\",\"color\"=\"blue\"]\n1 [\"label\"=\"a\",\"shape\"=\"ellipse\",\"fillcolor\"=\"red\",\"color\"=\"red\"]\n1 -- 3\n}\n"
+    @test (todota || todotb)
+    @test toDotFile(dfg, "something.dot") == nothing
+    Base.rm("something.dot")
+end
+
+@testset "Testing CRUD, return and Failures from a GraphsDFG" begin
+    global dfg
+    # fg to copy to
+    # creating a whole new graph with the same labels
+    T = typeof(dfg)
+    if T <: CloudGraphsDFG
+        dfg2 = CloudGraphsDFG{SolverParams}("localhost", 7474, "neo4j", "test",
+                                            "testUser", "testRobot", "testSession2",
+                                            nothing,
+                                            nothing,
+                                            IncrementalInference.decodePackedType,
+                                            IncrementalInference.rebuildFactorMetadata!,
+                                            solverParams=SolverParams())
+    else
+        dfg2 = T()
+    end
+
+    # Build a new in-memory IIF graph to transfer into the new graph.
+    iiffg = initfg()
```

Comment:
Please leave this in-tact. I want to continue extending tests to work between different graphs.

Reply:
we should rather create a new test for the _get duplicated graph.